ext/reflection/php_reflection.c

```diff
@@ -2112,18 +2112,25 @@ ZEND_METHOD(ReflectionGenerator, getExecutingGenerator)
 ZEND_METHOD(ReflectionParameter, __construct)
 {
 	parameter_reference *ref;
-	zval *reference, *parameter;
+	zval *reference;
+	zend_string *string_parameter = NULL;
+	zend_long position_parameter;
 	zval *object;
 	zval *prop_name;
 	reflection_object *intern;
 	zend_function *fptr;
 	struct _zend_arg_info *arg_info;
-	int position;
 	uint32_t num_args;
 	zend_class_entry *ce = NULL;
 	zend_bool is_closure = 0;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "zz", &reference, &parameter) == FAILURE) {
+	ZEND_PARSE_PARAMETERS_START(2, 2)
+		Z_PARAM_ZVAL(reference)
+		Z_PARAM_STR_OR_LONG(string_parameter, position_parameter)
+	ZEND_PARSE_PARAMETERS_END();
+
+	if (string_parameter == NULL && position_parameter < 0) {
+		zend_argument_value_error(2, "must be greater than or equal to 0");
```

Comment:
Why check this here rather than with the rest of the position_parameter check?

Reply:
That's a good point, will change this.